actors/evm/src/lib.rs

```diff
@@ -259,8 +264,31 @@ impl ActorCode for EvmContractActor {
                 let value = Self::storage_at(rt, cbor::deserialize_params(params)?)?;
                 Ok(RawBytes::serialize(value)?)
             }
+            Some(Method::InvokeContractReadOnly) => {
+                let BytesDe(params) = params.deserialize()?;
+                let value = Self::invoke_contract(
+                    rt,
+                    Method::InvokeContractReadOnly as u64,
+                    &params,
+                    true,
+                    None,
+                )?;
+                Ok(RawBytes::serialize(BytesSer(&value))?)
+            }
+            Some(Method::InvokeContractDelegate) => {
+                let params: DelegateCallParams = cbor::deserialize_params(params)?;
+                let value = Self::invoke_contract(
+                    rt,
+                    Method::InvokeContractDelegate as u64,
+                    &params.input,
+                    params.readonly,
+                    Some(params.code),
+                )?;
+                Ok(RawBytes::serialize(BytesSer(&value))?)
```

Comment:
This is ok for now, but we should really define all method logic in the actor methods themselves. Ideally, we'd only handle decoding/encoding here.